### PR TITLE
Make reset password page customizable, add README

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Custom Reset Password page content
- Updated README

#### Screenshots:
change password code

change password

### CODE REVIEW

**Comment by emclaug2** [comment]
This change allows for a ng-content substitutes where applicable; there's a good amount of customization in here now; this'll help a lot for i18n support.

**Comment by daileytj** [comment]
Looks great! This will certainly make i18n support much easier. My only comment on this is that the end user essentially has to null out our default @Input values in order to use the templates. Can you make it function such that when a template is present it takes precedence over our default values?

**Comment by emclaug2** [comment]
Yeah this will require a little bit of rework but it's worth it so that end users have more convenience.   I'll push out a change shortly.  We'll need to add a ViewChild for each ng-content override. Same goes with all other pages.

**Comment by emclaug2** [comment]
> Yeah this will require a little bit of rework but it's worth it so that end users have more convenience. I'll push out a change shortly. We'll need to add a ViewChild for each ng-content override. Same goes with all other pages.

Just to reiterate, this is going to be make things a lot more complicated and hard to read for us.  I'm going to start with this screen for a POC.

**Comment by emclaug2** [comment]
Bring on the change detection issues (thumbs-down)

**Comment by emclaug2** [comment]
@daileytj  We won't be able to easily do this with ng-content unless we start adding detectChanges after actions that should change the screen (when api calls resolve).  

This is getting a little overwhelming using ng-content. 

An alternative would be to have the user pass in custom template refs instead of a ng-content.  That way we can continue to do a simple check if the templateRef is empty onInit, instead of using that finicky isEmptyView function.

How's that sound?  It's different, it adds more lines that the user would have to provide, but it makes our logic a lot less crazy to manage.

**Comment by emclaug2** [comment]
For more context there's a new branch here that fixes the problem where uses have to remove the default string @input: https://github.com/pxblue/angular-workflows/tree/feature/custom-reset-password-cd-error-fixed

Trade-offs: 

- convenient for the user, not for us
- more template lines
- each @input variable needs a ViewChild
- changeDetection must be ran 2x after all promises resolve

**Comment by emclaug2** [comment]
Okay I've updated this branch to contain the default @input overrides; I think I've got the hang of this now.

File: login-workflow/src/pages/reset-password/reset-password.component.html [hunk 1]
```diff
@@ -53,14 +72,28 @@
...
         </mat-form-field>
     </form>
 </div>
 
 <div class="content" style="justify-content: center;" *ngIf="passwordResetSuccess">
-    <pxb-empty-state class="empty-state" [title]="successTitle" [description]="successDescription">
+    <pxb-empty-state class="empty-state">
         <mat-icon pxb-empty-icon class="empty-state-icon">check_circle</mat-icon>
+        <div pxb-title>
+            <div *ngIf="isEmpty(resetSuccessTitleEl)">{{resetSuccessTitle}</div>
```
Comment (emclaug2) [inline login-workflow/src/pages/reset-password/reset-password.component.html:86]: This is fixed on my pc, not pushed out yet.

Comment (emclaug2) [reply login-workflow/src/pages/reset-password/reset-password.component.html:86]: {{ missingRightCurly }

**Comment by huayunh** [comment]
I am not here to do the PR review; just want to drop a comment that the screenshot in the PR description totally made my day.

**Review by daileytj — APPROVED** [review]
Aside from the missing curly brace this looks good to me.

**Comment by emclaug2** [comment]
Okay cool I'm going to update the contact support page to use the same approach too.

**Pull request merged**